jsettlers.mapcreator/src/jsettlers/mapcreator/main/DataTester.java

```diff
@@ -49,21 +52,48 @@
 	private final ErrorList errorList;
 	private ArrayList<Error> errors = new ArrayList<Error>();
 
+	/**
+	 * Thread for testing
+	 */
+	private Thread thread;
+
+	/**
+	 * Running flag
+	 */
+	private boolean running = true;
+
 	public DataTester(MapData data, TestResultReceiver receiver) {
 		this.data = data;
 		this.receiver = receiver;
 		errorList = new ErrorList();
 	}
 
-	public void start() {
-		new Thread(this, "data tester").start();
+	public synchronized void start() {
+		running = true;
+		this.thread = new Thread(this, "data tester");
+		thread.start();
+	}
+
+	/**
+	 * Release all resources
+	 */
+	public void dispose() {
+		running = false;
+		synchronized (this) {
+			notifyAll();
+		}
+		try {
+			thread.join(100);
```

Comment:
Currently the editor is always quit with System.exit, because the re are a lot of threads which aren't stopped.
But you're right... It makes no sense as it is currently, i add a thread.stop() to kill after, OK?
```java
			thread.join(100);
			thread.stop();
```
Not the nice solution, but its currently anyway only at the end, and the thread should have enough time to quit.